src/completers/completeState/index.js

```diff
@@ -1,22 +1,44 @@
-import { isStringArray, isValidObject } from '../../utils/typeUtils';
+import { isStringArray, isValidObject, isObjectArray } from '../../utils/typeUtils';
 
-// Given a defaultState, it populates that state with ${key}Loading and ${key}Error
-function completeState(defaultState, ignoredTargets = []) {
-  if (!isValidObject(defaultState)) {
-    throw new Error('Expected an object as a state to complete');
+const isValidCustomCompleter = (targets, completers) =>
+  isStringArray(targets) && isValidObject(completers);
+
+export const customCompleter = (target, completers = {}) =>
+  Object.keys(completers)
+    .map(completer => ({ [`${target}${completer}`]: completers[completer] }))
+    .reduce((acc, value) => ({ ...acc, ...value }), {});
+
+function completeState({ description, targetCompleters = [], ignoredTargets = [] }) {
+  if (!isValidObject(description)) {
+    throw new Error('Expected an object as a description');
   }
   if (!isStringArray(ignoredTargets)) {
     throw new Error('Expected an array of strings as ignored targets');
   }
+  if (!isObjectArray(targetCompleters)) {
+    throw new Error('Expected an array of objects as a target completers');
+  }
 
-  const completedState = { ...defaultState };
-  Object.keys(defaultState)
+  const primaryState = Object.keys(description)
     .filter(key => ignoredTargets.indexOf(key) === -1)
-    .forEach(key => {
-      completedState[`${key}Loading`] = false;
-      completedState[`${key}Error`] = null;
-    });
-  return completedState;
+    .reduce((acc, key) => ({
+      ...acc,
+      [`${key}Loading`]: false,
+      [`${key}Error`]: null
+    }), {});
+
+  const customCompleters = targetCompleters
+    .map(({ completers, targets }) => {
+      if (!isValidCustomCompleter(targets, completers)) {
+        throw new Error('Expected an objects with targets as string array and completers as valid object');
+      }
+      return targets
+        .map(target => customCompleter(target, completers))
```

Comment:
🤔 I think it's better to let the user decide how to complete the state.
For example,
```
completeState({targetCompleters:[{
targets:['unT', 'otroT'],
completer: target => ({
[`${target}Bla`]: 1,
[`${target}Bleh`]: 2
})
}]})
```

```
{
unTbla:1,
unTBleh: 2,
otroTbla:1,
otroTBleh: 2
}
```

Reply:
Don't you think the result should be something like 
```javascript
{
  unT: 'Some Value',
  unTbla:1,
  unTBleh: 2,
  otroT: 'anothher value',
  otroTbla:1,
  otroTBleh: 2
}
```

Reply:
As we talked yesterday, I don't think we should add the `unT: 'Some Value',`. If the user wants that, he can put it on the completer function